src/scripts/components/VariableMessage.jsx

```diff
@@ -18,24 +20,43 @@ import {
 } from '../constants';
 
 class VariableMessage extends React.Component {
+  currentDraggedSpanVariable = '';
+  isCategoryMobileViewVisible = false;
   state = {
     message: '',
     available: [],
+    variables: [],
+    variablesOfCategory: [],
+    categories: [],
+    selectedCategory: '',
   };
 
   componentWillMount() {
+    const toUpdateState = {
+      categories: [],
+    };
+
+    if (this.props.isCategoryAvailable) {
+      this.isCategoryMobileViewVisible = window.outerWidth <= 767; //  for mobile view
```

Comment:
This relates to my previous comment it will only display the component properly on first load but not react to any changes thereafter.